pkg/login/social/connectors/generic_oauth.go

```diff
@@ -320,11 +319,11 @@ func (s *SocialGenericOAuth) UserInfo(ctx context.Context, client *http.Client,
 	}
 
 	if !s.isTeamMember(ctx, client) {
-		return nil, errors.New("user not a member of one of the required teams")
+		return nil, &SocialError{"User not a member of one of the required teams"}
```

Comment:
This follows what the current setup for other OAuth providers is. See below:

https://github.com/grafana/grafana/blob/721c50a304588ebd7cea76e301ec0f68a5a55d68/pkg/login/social/connectors/common.go#L41